### ISSUE TITLE
np.searchsorted doesnt work with DatetimeIndex

### ISSUE THREAD

**Issue by Biarys** [numpy/numpy#16175]
np.searchsorted doesnt work when array is pd.DatetimeIndex. Instead it returns either the datetime (instead of index location. Tried recreating in Jupyter notebook) or datetime with appended location in the end (in VS Code where I was running it. Please see the screenshot).

### Reproducing code example:

```python
import numpy as np
import pandas as pd
idx = pd.DatetimeIndex(['1976-08-24', '1976-08-25', '1976-08-26', '1976-08-27'])
date = pd.Series(idx[0])
res = np.searchsorted(idx, date)
print(res)
# outputs
# DatetimeIndex(['1970-01-01'], dtype='datetime64[ns]', freq=None) # wrong date

idx2 = pd.Index(idx, dtype=object)
date2 = idx2[2]
res2 = np.searchsorted(idx2, date2)
print(res2)
# outputs 
# 2
```
[https://ibb.co/5kSYqQV](url) <- screenshot

### Error message:
No error

### Numpy/Python version information:

1.18.1 3.7.3 (default, Apr 24 2019, 15:29:51) [MSC v.1915 64 bit (AMD64)]

**Comment by rossbar** [comment]
Related: #15770 and #15712 .

Note that your example works as expected if you explicitly operate on numpy arrays instead of pandas structures:

```python
>>> idx = pd.DatetimeIndex(['1976-08-24', '1976-08-25', '1976-08-26', '1976-08-27'])
>>> date = pd.Series(idx[0])
>>> np.searchsorted(idx.to_numpy(), date.to_numpy())
array([0])     # Proper index
```

**Comment by seberg** [comment]
This works for me locally. I believe this is a fixed pandas issue. Not sure what or when exactly pandas fixed it though. Will close it anyway for now (although it may be nice if you can double check that this was really a pandas issue).

**Comment by seberg** [comment]
Sorry, I did not recall having an up to date dev-version installed in that environment. So this is probably fixed, but fixed in pandas master and may not yet be released.

**Comment by rossbar** [comment]
Not yet fixed as of pandas v1.0.3, but it looks like it's taken care of in the dev (v1.1) version
